modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogServiceImpl.java

```diff
@@ -863,4 +654,197 @@ private Catalog applyUpdateFinal(Catalog catalog, VersionedUpdate update) {
                 catalog.schemas()
         );
     }
+
+    private static CatalogSchemaDescriptor getSchema(Catalog catalog, @Nullable String schemaName) {
+        schemaName = Objects.requireNonNullElse(schemaName, DEFAULT_SCHEMA_NAME);
+
+        return Objects.requireNonNull(catalog.schema(schemaName), "No schema found: " + schemaName);
```

Comment:
Why do we throw an NPE when there is no such schema, but a specific exception when there is no such zone/table?

Reply:
fix it